### PR TITLE
Fix Node.js deprecated install script

### PR DESCRIPTION
**Pull request by snaselj**
# Closes: NaN

## What's Changed

- Fixed Dockerfile Node.js `setup_XX.x` [deprecated script](https://github.com/nodesource/distributions#new-update-%EF%B8%8F).
- Split Dockerfile RUN into separate apt, npm and pip instructions for better readability.

## Justification

There was 1 minute timeout during container build:

```
#12 14.37 ================================================================================
#12 14.37 ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
#12 14.37 ================================================================================
#12 14.37 
#12 14.37                            SCRIPT DEPRECATION WARNING                    
#12 14.37 
#12 14.37   
#12 14.37   This script, located at https://deb.nodesource.com/setup_X, used to
#12 14.37   install Node.js is deprecated now and will eventually be made inactive.
#12 14.37 
#12 14.37   Please visit the NodeSource distributions Github and follow the
#12 14.37   instructions to migrate your repo.
#12 14.37   https://github.com/nodesource/distributions
#12 14.37 
#12 14.37   The NodeSource Node.js Linux distributions GitHub repository contains
#12 14.37   information about which versions of Node.js and which Linux distributions
#12 14.37   are supported and how to install it.
#12 14.37   https://github.com/nodesource/distributions
#12 14.37 
#12 14.37 
#12 14.37                           SCRIPT DEPRECATION WARNING
#12 14.37 
#12 14.37 ================================================================================
#12 14.37 ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
#12 14.37 ================================================================================
#12 14.37 
#12 14.37 TO AVOID THIS WAIT MIGRATE THE SCRIPT
#12 14.37 Continuing in 60 seconds (press Ctrl-C to abort) ...
#12 14.37 
#12 74.37 
#12 74.37 ## Installing the NodeSource Node.js 18.x repo...
```

### CODE REVIEW

**Comment by gsnider2195** [comment]
Please rebase to `develop`

File: docker/Dockerfile [hunk 1]
```diff
@@ -49,25 +49,34 @@ ENV PYTHONUNBUFFERED=1 \
...
+RUN --mount=type=cache,target="/var/cache/apt",sharing=locked \
+    --mount=type=cache,target="/var/lib/apt/lists",sharing=locked \
+    apt-get update && \
+    apt-get install --no-install-recommends -y ca-certificates curl gnupg && \
+    mkdir -p /etc/apt/keyrings && \
+    curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
+    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list && \
     apt-get update && \
     apt-get upgrade -y && \
     apt-get install --no-install-recommends -y curl && \
```
Comment (gsnider2195) [inline docker/Dockerfile:67]: Remove curl install here? (already installed on line 61)

File: docker/Dockerfile [hunk 2]
```diff
@@ -49,25 +49,33 @@ ENV PYTHONUNBUFFERED=1 \
...
+    mkdir -p /etc/apt/keyrings && \
+    curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
+    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list && \
     apt-get update && \
     apt-get upgrade -y && \
-    apt-get install --no-install-recommends -y curl && \
-    curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
     apt-get install --no-install-recommends -y git mime-support libxml2 libmariadb3 openssl nodejs && \
-    apt-get autoremove -y && \
-    apt-get clean all && \
```
Comment (gsnider2195) [inline docker/Dockerfile:65]: Is this line not required?

Comment (snaselj) [reply docker/Dockerfile:65]: > Is this line not required?

It's not necessary, as apt temporary folders are mounted as cache, lines 58, 59

Comment (snaselj) [reply docker/Dockerfile:65]: > Same question for the rest of the cleanup lines removed below

The same, solved by mounting apt temporary folders to Docker cache.

Comment (snaselj) [reply docker/Dockerfile:65]: > We need the `rm -rf /tmp/tmp*`. It tries to clean up that directory as the nautobot user later in the `python-dependencies` stage and fails

`/mnt` is mounted as a Docker cache at line 77, so the line 78: `RUN pip install ...` instruction do not leave any files in /tmp on the image.

**Comment by snaselj** [comment]
Merged with the latest develop to include venv removal.

File: docker/Dockerfile (hunk 2 above)
Comment (gsnider2195) [reply docker/Dockerfile:65]: This build was failing for me when trying to delete a `/tmp/tmp<random string>` file yesterday but seems to be working now. Maybe I had a bad cache.

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**